packages/backend/tests/unit/workers/metrics.test.ts

```
@@ -0,0 +1,379 @@
+    ),
+  }));
+
+  mock.module('bullmq', () => ({
+    Worker: class MockWorker {
+      constructor(_name: string, _processor: (job: unknown) => Promise<unknown>) {}
+      on(event: string, handler: Handler) {
+        // Append rather than overwrite — workers attach multiple `failed`
+        // listeners (e.g. hash-list-parser has one for metrics and one for
+        // DB cleanup). Capturing only the last listener masks regressions
+        // where the metrics listener gets unregistered.
+        (capturedHandlers[event] ??= []).push(handler);
```

Biome (2.4.15)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)